demisto_sdk/commands/content_graph/tests/get_dependencies_test.py

```diff
@@ -0,0 +1,583 @@
+        |  ------------------------------------------                    | |
+        |       DEPENDS_ON     |                                         | |
+        |      (optionally)     ------------------------------------------ |     DEPENDS_ON
+        |                                     DEPENDS_ON                   | (mandatorily, is_test)
+        |                                    (mandatorily)                 |
+         ------------------------------------------------------------------
+
+    Args:
+        graph_repo (Repo): the content repo to work with.
+    """
+    pack1 = graph_repo.create_pack("SamplePack1")
+    pack1.set_data(
+        dependencies={
+            "SamplePack3": {"mandatory": False, "display_name": "SamplePack3"},
+        }
+    )
+    pack2 = graph_repo.create_pack("SamplePack2")
+    pack2.set_data(
+        dependencies={
+            "SamplePack3": {"mandatory": False, "display_name": "SamplePack3"}
+        }
+    )
+    graph_repo.create_pack("SamplePack3").set_data(
+        dependencies={
+            "SamplePack4": {"mandatory": False, "display_name": "SamplePack4"}
+        }
+    )
+    pack4 = graph_repo.create_pack("SamplePack4")
+    pack4.set_data(
+        dependencies={
+            "SamplePack2": {"mandatory": True, "display_name": "SamplePack2"},
+            "SamplePack5": {"mandatory": True, "display_name": "SamplePack5"},
+        }
+    )
+    graph_repo.create_pack("SamplePack5").set_data(hidden=True)
+
+    pack2.create_script("SampleScript2")
+    pack2.create_integration("SampleIntegration2").set_commands(["pack2-test-command"])
+    pack1.create_script("SampleScript").yml.update(
+        {"dependson": {"must": ["SampleScript2", "pack2-test-command"]}}
+    )
+
+    pack4_test_playbook = pack4.create_test_playbook("SampleTestPlaybook")
+    pack4_test_playbook.add_default_task("SampleScript")
+
+
+def compare(
+    result: list,
+    expected_list: list,
+    dependency_pack: str,
+    mandatory_only: bool,
+    include_tests: bool,
+    show_reasons: bool,
+) -> None:
+    assert len(result) == len(expected_list)
+    result.sort(key=lambda r: r["object_id"])
+    expected_list.sort(key=lambda r: r["object_id"])
+    for actual, expected in zip(result, expected_list):
+        assert actual.get("mandatorily") == expected.get("mandatorily")
+        assert actual.get("paths")[0].get("is_test") == expected.get("is_test")
+        assert actual.get("minDepth") == expected.get("minDepth")
+
+        if mandatory_only:
+            assert actual.get("mandatorily")
+
+        if not include_tests:
+            assert not any([path.get("is_test") for path in actual.get("paths")])
+
+        if dependency_pack:
+            assert actual.get("object_id") == dependency_pack
+
+        if show_reasons:
+            assert actual.get("reasons") == expected.get("reasons")
+        else:
+            assert "reasons" not in actual
+
+
+class TestGetRelationships:
```

Comment:
as decided at the team level, let's split this into multiple test cases
regardless - great tests!

Reply:
you're still using parametrize, let's discuss if my note here was not clear